include/HubFramework/HUBComponentModel.h

```diff
@@ -229,6 +229,11 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, strong, nullable, readonly) NSArray<id<HUBComponentModel>> *children;
 
 /**
+ *  The index path of this component in the `HUBComponentModel` hierarchy.
+ */
+@property (nonatomic, strong, readonly) NSIndexPath *indexPath;
```

Comment:
We should add a note that this property is computed, so that it's not accessed multiple times within the same block of code. Alternatively, we could cache the value. It's up to you :)